util/dvsim/doc/testplanner.md

```diff
@@ -0,0 +1,258 @@
+---
+title: "Testplanner tool"
+---
+
+`testplanner` is a Python based tool for parsing testplans written in Hjson format into a data structure that can be used for:
+* Expanding the testplan inline within the DV document as a table
+* Annotating the simulation results with testplan entries for a document driven DV execution
+
+Please see [DV methodology]({{< relref "doc/ug/dv_methodology/index.md#documentation" >}}) for more details on the rationale and motivation for writing and maintaining testplans in a machine-parsable format (`Hjson`).
+This document will focus on the anatomy of an Hjson testplan, the list of features supported and some of the ways of using the tool.
+
+## Hjson testplan
+
+A testplan consists of a list of planned tests (testpoints) and a list of planned functional coverage items (covergroups).
+
+### Testpoints
+
+A testpoint is an entry in the testplan representing a planned test.
+Each testpoint maps one-to-one to a unique feature of the design.
+Additionally, a testpoint for each of the [key areas of focus]({{< relref "doc/ug/dv_methodology/index.md#key-test-focus-areas" >}}) (whichever ones are applicable) is also captured in the testplan.
+
+The following attributes are used to define each testpoint, at minimum:
+* **name: testpoint name**
+
+    This is a single `lower_snake_case` string that succinctly describes the intended feature being tested.
+    As an example, a smoke test, which is typically the first test written for a new testbench would simply be called `smoke`.
+    A testpoint that focuses on erasing a page within a `flash_ctrl` bank would be called `page_erase`.
+    The recommended naming convention to follow is `<main_feature>_<sub_feature>_<sub_sub_feature_or_type>_...`.
+    This is no need to suffix (or prefix) the testpoint name with "test".
+
+* **milestone: targeted verification milestone**
+
+    This is either `V1`, `V2` or `V3`.
+    It helps us track whether all of the testing requirements of a milestone have been achieved.
+
+* **desc: testpoint description**
+
+    A multi-line string that briefly describes the intent of the test.
+    It is recommended, but not always necessary to add a high level goal, stimulus, and the checking procedure so that the reader gets a clear idea of what and how the said feature is going to be tested.
+
+    Full [Markdown]({{< relref "doc/rm/markdown_usage_style" >}}) syntax is supported when writing the description.
+
+* **tests: list of written test(s) for a testpoint**
+
+    The testplan is written in the initial work stage of the verification [life-cycle]({{< relref "doc/project/development_stages#hardware-verification-stages" >}}).
+    Later, when the DV engineer writes the tests, it may not map one-to-one to a testpoint - it may be possible that a written test satisfactorilly addresses multiple testpoints; OR it may also be possible that a testpoint needs to be split into multiple smaller tests.
```

Comment:
Nit: Singular/plural mismatch. Maybe "... writes the tests, they may not map..." ?